src/ert/config/queue_config.py

```diff
@@ -17,123 +19,159 @@
-        "QDEL_CMD",
-        "QSTAT_OPTIONS",
-        "QUEUE",
-        "CLUSTER_LABEL",
-        "JOB_PREFIX",
-        "PROJECT_CODE",
-    ],
-    "LOCAL": ["PROJECT_CODE"],
-    "GENERIC": [],
-}
-queue_positive_int_options: Mapping[str, List[str]] = {
-    "LSF": [
-        "MAX_RUNNING",
-    ],
-    "SLURM": [
-        "MAX_RUNNING",
-    ],
-    "TORQUE": [
-        "NUM_NODES",
-        "NUM_CPUS_PER_NODE",
-        "MAX_RUNNING",
-    ],
-    "LOCAL": ["MAX_RUNNING"],
-    "GENERIC": ["MAX_RUNNING"],
-}
-queue_positive_number_options: Mapping[str, List[str]] = {
-    "LSF": ["SUBMIT_SLEEP"],
-    "SLURM": ["SUBMIT_SLEEP", "SQUEUE_TIMEOUT", "MAX_RUNTIME"],
-    "TORQUE": ["SUBMIT_SLEEP", "QUEUE_QUERY_TIMEOUT"],
-    "LOCAL": ["SUBMIT_SLEEP"],
-    "GENERIC": ["SUBMIT_SLEEP"],
-}
-queue_bool_options: Mapping[str, List[str]] = {
-    "LSF": [],
-    "SLURM": [],
-    "TORQUE": ["KEEP_QSUB_OUTPUT"],
-    "LOCAL": [],
-    "GENERIC": [],
-}
-queue_memory_options: Mapping[str, List[str]] = {
-    "LSF": [],
-    "SLURM": ["MEMORY_PER_CPU", "MEMORY"],
-    "TORQUE": ["MEMORY_PER_JOB"],
-    "LOCAL": [],
-    "GENERIC": [],
-}
+NonEmptyString = Annotated[str, pydantic.StringConstraints(min_length=1)]
+
+
+@pydantic.dataclasses.dataclass(config={"extra": "forbid", "validate_assignment": True})
+class QueueOptions:
+    max_running: pydantic.NonNegativeInt = 0
+    submit_sleep: pydantic.NonNegativeFloat = 0.0
+    project_code: Optional[str] = None
+
+    @staticmethod
+    def create_queue_options(
+        queue_system: QueueSystem,
+        options: Dict[str, Any],
+        is_selected_queue_system: bool,
+    ) -> Optional[QueueOptions]:
+        lower_case_options = {key.lower(): value for key, value in options.items()}
+        try:
+            if queue_system == QueueSystem.LSF:
+                return LsfOptions(**lower_case_options)
+            elif queue_system == QueueSystem.SLURM:
+                return SlurmOptions(**lower_case_options)
+            elif queue_system == QueueSystem.TORQUE:
+                return TorqueOptions(**lower_case_options)
+            elif queue_system == QueueSystem.LOCAL:
+                return LocalOptions(**lower_case_options)
+            elif queue_system == QueueSystem.GENERIC:
+                return QueueOptions(**lower_case_options)
+        except pydantic.ValidationError as exception:
+            for error in exception.errors():
+                _throw_error_or_warning(
+                    f"{error['msg']}. Got input '{error['input']}'.",
+                    error["input"],
+                    is_selected_queue_system,
+                )
+            return None
+
+    def add_global_queue_options(self, config_dict: ConfigDict) -> None:
+        for generic_option in fields(QueueOptions):
+            if (
+                generic_value := config_dict.get(generic_option.name.upper(), None)  # type: ignore
+            ) and self.__dict__[generic_option.name] == generic_option.default:
```

Comment:
what is this other constrain: `self.__dict__[generic_option.name] == generic_option.default:`?

Reply:
It is to satisfy `test_global_config_key_does_not_overwrite_queue_options`